packages/framework/block-std/src/view/element/edgeless-block-element.ts

```diff
@@ -0,0 +1,172 @@
+import type { BlockModel } from '@blocksuite/store';
+
+import { nothing } from 'lit';
+
+import type { SerializedXYWH } from '../../edgeless/types.js';
+import type { BlockService } from '../../service/index.js';
+
+import { BlockElement } from './block-element.js';
+
+export const edgelessElementSymbol = Symbol('edgelessElement');
+
+export abstract class EdgelessBlockElement<
+  EdgelessRootService extends BlockService = BlockService,
+  Model extends BlockModel = BlockModel,
+  Service extends BlockService = BlockService,
+  WidgetName extends string = string,
+> extends BlockElement<Model, Service, WidgetName> {
+  [edgelessElementSymbol] = true;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.style.position = 'absolute';
+  }
+
+  getRenderingRect() {
+    const { xywh } = this.model as BlockModel<{
+      xywh: SerializedXYWH;
+      index: string;
+    }>;
+
+    if (!xywh) {
+      throw new Error('Edgeless block should have at least `xywh` property.');
+    }
+
+    const [x, y, w, h] = JSON.parse(xywh);
+
+    return { x, y, w, h, zIndex: this.toZIndex() };
+  }
+
+  override renderBlock() {
+    const { xywh, index } = this.model as BlockModel<{
```

Comment:
You can use: 
```ts
const { xywh$, index$ } = this.model;

if (!xywh$.value || !index$.value) { ... }
```

This reactive prop will be updated automatically without the subscription to `propsUpdated`.

Reply:
Got it. I will refine these part in the next PRs.